src/server.rs

```diff
@@ -957,6 +970,42 @@ impl Server {
                     if self.in_copy_mode {
                         self.in_copy_mode = false;
                     }
+                    // TODO: consider logging a warning here
+
+                    if self.prepared_statement_enabled {
```

Comment:
Let's say we have a statement that uses `select *` on a table but a new column is added after we prepared it. PG will send an error if we try to use this statement, this change tries to identify this type of error and drop the server connection (and all prepared statements) to force re-prepares. Clients will still see errors but this will help clean up the pool.